### PR TITLE
🚧 Admin style v3: Add ability for non super-admin user to see products_v3 page

### LINKED ISSUE #11268: [BUU] Allow to activate feature toggle for users outside of superadmin + one actor only

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

- Closes #11268 

##### With feature `admin_style_v3` deactivated
No user should have acess

##### With feature `admin_style_v3` activated (global or per actor)
- As a non super-admin user, check that you've access to `/admin/products_v3` page if feature is activated for you (global or per actor and if you also have access to `/admin/products` page)
- As a super-admin user, check that you've access to `/admin/products_v3` page if feature is activated for you (global or per actor)

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by rioug — APPROVED** [review]
Looks good :+1:

File: app/models/spree/ability.rb [hunk 1]
```diff
@@ -191,6 +191,8 @@ def add_product_management_abilities(user)
         OpenFoodNetwork::Permissions.new(user).managed_product_enterprises.include? product.supplier
       end
 
+      can [:admin, :index], :products_v3
```
Comment (dacook) [inline app/models/spree/ability.rb:194]: I have to admit I need to check the docs each time I look at the ability file! I don't understand what `:admin` is for, but tried it out and can see that this works for the `index` action. 

I also don't understand how the existing products page is working and why we need to add this line! The existing rule above (`can`...`:index`...`Spree::Product`) is validated on a specific `product`, but when showing an index there's no single product to validate, and we manage access in the query instead.

In any case, I can see that this works 👍

**Review by dacook — APPROVED** [review]
This works, although I don't understand completely!

Unfortunately the product loading is broken on my envrionment, with no errors to explain (I think this is a problem on master at the moment, which is fixed in https://github.com/openfoodfoundation/openfoodnetwork/pull/11163).

So this looks good to me, and we can test that you can access the page. 
But it may fail to load any products due to another problem, so I'd suggest we don't test until it's been rebased on the above fix.

File: app/models/spree/ability.rb (hunk 1 above)
Comment (jibees) [reply app/models/spree/ability.rb:194]: I must admit that it's not totally clear to me as well. Specially this `:admin`. But I understand why I should add this line. 
This is due to 

https://github.com/openfoodfoundation/openfoodnetwork/blob/cb4aa946de96de017e5e6a361ee952f265c90a3a/app/controllers/spree/admin/base_controller.rb#L45-L57

when requesting `ProductsV3Controller`, we go through the `else`, and then `record` is finally equal to `:products_v3`

**Comment by jibees** [comment]
> (I think this is a problem on master at the moment, which is fixed in #11163).

Yes, due to the deletion of `available_on` attribute on product

**Comment by jibees** [comment]
Due to "circular dependency", @dacook and I took the responsibility to cherry pick the commit of this one and to put into https://github.com/openfoodfoundation/openfoodnetwork/pull/11163 

--> https://github.com/openfoodfoundation/openfoodnetwork/pull/11163/commits/2a81a9acfb2cadec1eb6aeeb8c8199e479d3dde4

Closing here then.

**Pull request closed without merging**